### PR TITLE
Python3 spec file

### PR DESCRIPTION
**Pull request by davidlange6**
Thanks to @smuzaffar for a working python3 spec.

in addition, we've added a PYTHON27PATH variable to eventually remove our usage of PYTHONPATH (so possible to support python2 and 3 for some period)

### CODE REVIEW

**Comment by davidlange6** [comment]
please test

**Comment by davidlange6** [comment]
(most of cmssw-tool-conf is rebuild in my area by now-so the test should be ok) - i would propose to merge both sets of python changes before the 2300 tonight

File: python3.spec [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+find %i -name '*.pyo' -exec rm {} \;
+
+# Remove documentation, examples and test files.
+rm -rf %{i}/share %{i}/lib/python*/test %{i}/lib/python%{pythonv}/distutils/tests %{i}/lib/python%{pythonv}/lib2to3/tests
+
+echo "from os import environ" > %i/lib/python%{pythonv}/sitecustomize.py
+echo "if 'PYTHON3PATH' in environ:" >> %i/lib/python%{pythonv}/sitecustomize.py
+echo "   import sys,site" >> %i/lib/python%{pythonv}/sitecustomize.py
+echo "   for p in environ['PYTHON3PATH'].split(':'):">> %i/lib/python%{pythonv}/sitecustomize.py
+echo "      site.addsitedir(p)" >> %i/lib/python%{pythonv}/sitecustomize.py
```
Comment (smuzaffar) [inline python3.spec]: should we change this logic to something like
```
from os import environ
if 'PYTHON3PATH' in environ:
  import sys
  sys.path =environ['PYTHON3PATH'].split(':') + sys.path
```
this way without any changes in SCRAM or buildrules (to dump .pth files) we can just update toolfiles to replace PYTHONPATH with PYTHON3PATH

**Comment by davidlange6** [comment]
please test

**Pull request merged**